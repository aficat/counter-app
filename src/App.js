import React, { Component } from 'react';
import Button from 'react-bootstrap/Button';
import { Jumbotron } from 'react-bootstrap';

class App extends Component {
  constructor() {
    super();
    this.state = {
      count: 0,
    }
  }

  makeIncrementer = amount => () =>
    this.setState(prevState => ({
      count: prevState.count + amount,
    }));

  increment = this.makeIncrementer(1);

  decrement = this.makeIncrementer(-1);

  render() {
    return (
      <div className="container" style={{ paddingTop: 50 }} align="center">
        <Jumbotron>
          <h1>Counter App</h1>
          <h4>Count: {this.state.count}</h4>
          <br />
          <Button id="increment" onClick={this.increment}>Increment</Button>
          &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
          <Button id="decrement" onClick={this.decrement}>Decrement</Button>
        </Jumbotron>
      </div>
    )
  }
}

export default App;