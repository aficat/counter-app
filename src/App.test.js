import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('App component', () => {
  it('starts with a count of 0', () => {
    const wrapper = shallow(<App />);
    const text = wrapper.find('h1').text();
    expect(text).toEqual('Counter App');
  });

  it('starts with a count of 0', () => {
    const wrapper = shallow(<App />);
    const text = wrapper.find('h4').text();
    expect(text).toEqual('Count: 0');
  });

  it('increments count by 1 when the increment button is clicked', () => {
    const wrapper = shallow(<App />);
    const incrementBtn = wrapper.find('#increment');
    incrementBtn.simulate('click');
    const text = wrapper.find('h4').text();
    expect(text).toEqual('Count: 1');
  });

  it('decrements count by 1 when the decrement button is clicked', () => {
    const wrapper = shallow(<App />);
    const decrementBtn = wrapper.find('#decrement');
    decrementBtn.simulate('click');
    const text = wrapper.find('h4').text();
    expect(text).toEqual('Count: -1');
  });
});

describe('Addition', () => {
  it('knows that 2 and 2 make 4', () => {
    expect(2 + 2).toBe(4);
  });

  test('knows that 2 and 2 make 4', () => {
    expect(2 + 2).toBe(4);
  });
});

it('matches the snapshot', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});