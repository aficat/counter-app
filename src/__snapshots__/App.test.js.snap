// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  align="center"
  className="container"
  style={
    Object {
      "paddingTop": 50,
    }
  }
>
  <div
    className="jumbotron"
  >
    <h1>
      Counter App
    </h1>
    <h4>
      Count: 
      0
    </h4>
    <br />
    <button
      className="btn btn-primary"
      disabled={false}
      id="increment"
      onClick={[Function]}
      type="button"
    >
      Increment
    </button>
         
    <button
      className="btn btn-primary"
      disabled={false}
      id="decrement"
      onClick={[Function]}
      type="button"
    >
      Decrement
    </button>
  </div>
</div>
`;
